backend/stylists/app/Http/Controllers/TipController.php

```diff
@@ -0,0 +1,257 @@
+        $this->initFilters();
+        
+        $view_properties = array(
+            'stylists' => $this->createdBy,
+            'statuses' => $this->statuses,
+            'genders' => $this->genders,
+            'occasions' => $this->occasions,
+            'body_types' => $this->body_types,
+            'budgets' => $this->budgets,
+           'age_groups' => $this->age_groups
+        );
+        
+        foreach ($this->filter_ids as $filter) {
+            $view_properties[$filter] = $request->has($filter) && $request->input($filter) !== "" ? intval($request->input($filter)) : "";
+        }
+        $view_properties['stylist_id'] = $request->has('stylist_id') && $request->input('stylist_id') !== "" ? intval($request->input('stylist_id')) : "";
+        
+        $entity_nav_tabs = array(
+            EntityType::CLIENT
+        );
+
+        $view_properties['entity_type_names']= array(
+            EntityTypeName::CLIENT
+        );
+        
+        $remove_deleted_looks = '1=1';
+        $paginate_qs = $request->query();
+        unset($paginate_qs['page']);
+        
+        $tips  = Tip::where($this->where_conditions)
+                       ->whereRaw($this->where_raw)
+                       ->whereRaw($remove_deleted_looks)
+                       ->orderBy('id', 'desc')
+                       ->simplePaginate($this->records_per_page)
+                       ->appends($paginate_qs);
+         
+         $view_properties['tips'] = $tips;
+        
+        $view_properties['nav_tab_index'] = '0';
+        $user_data = Auth::user();
+        
+        $view_properties['search'] = $request->input('search');
+        $view_properties['exact_word'] = $request->input('exact_word');
+        $view_properties['from_date'] = $request->input('from_date');
+        $view_properties['to_date'] = $request->input('to_date');
+
+        $view_properties['min_price'] = $request->input('min_price');
+        $view_properties['max_price'] = $request->input('max_price');
+        
+        $view_properties['app_sections'] = AppSections::all();
+        $view_properties['logged_in_stylist_id'] = $user_data->id;
+        $view_properties['popup_entity_type_ids'] = $entity_nav_tabs;
+        $view_properties['entity_type_to_send'] = EntityType::LOOK;
+        $view_properties['recommendation_type_id'] = RecommendationType::MANUAL;
+        $view_properties['is_owner_or_admin'] = Auth::user()->hasRole('admin');
+        
+        
+        return view('tip.list', $view_properties);
+
+    }
+  
+    public function getView()
+    {
+        $tip                = Tip::find($this->resource_id);
+        $view_properties    = [];
+        
+        if ($tip) {
+            $products           = $tip->products;
+            $status             = Status::find($tip->status_id);
+            
+            $view_properties    = array('tip' => $tip, 'products' => $products, 'stylist' => $tip->stylist, 'status' => $status);
```

Comment:
no need to pass ('products' => $products, 'stylist' => $tip->stylist, 'status' => $status) parameters as these are already associated with the tip object.
in view, at the place of products, use $tip->products. same for the other parameters